templates/widget/__tests__/widget.test.ts

```diff
@@ -0,0 +1,33 @@
+import 'reflect-metadata';
+import { MessageClient, MessageService } from '@theia/core';
+import { ContainerModule, Container } from 'inversify';
+import { MockMessageClient, MockMessageService } from './mock-objects/mock-message-service';
+import { <%= params.extensionPrefix %>Widget } from '../<%= params.extensionPath %>-widget';
+import { render } from '@testing-library/react'
+
+describe('widget extension unit tests', () => {
+
+    let widget: any;
```

Comment:
We should include the actual typings else we have no autocompletion and it leads to odd code later on.

Reply:
Yes, did that, but had to adapt the signature of widget.

Reply:
@JonasHelming please note that the following can remain `protected` and we can instead update the test:

```ts
protected displayMessage(): void {
    this.messageService.info('Congratulations: FooBar Widget Successfully Created!');
}
```

```ts
it('should inject \'MessageService\'', () => {
    const spy = jest.spyOn(widget as any, 'displayMessage')
    widget['displayMessage']();
    expect(spy).toBeCalled();
});
```

I'll leave the decision up to you, but I feel as though the `displayMessage` should remain `protected` as it should not allow outside callers when developers are modifying the template.

Reply:
Yeah, was torn between two options, now took yours...